api/tests/routes/public/collective/endpoints/simulate_adage_steps/test_bookings.py

```diff
@@ -379,3 +379,49 @@ def cancel_booking(self, client, booking_id, status_code, json_error=None):
         if json_error:
             for key, msg in json_error.items():
                 assert response.json.get(key) == msg
+
+
+class RepayCollectiveBookingTest(PublicAPIRestrictedEnvEndpointHelper):
+    endpoint_url = "/v2/collective/adage_mock/bookings/{booking_id}/repay"
+    endpoint_method = "post"
+    default_path_params = {"booking_id": 1}
+
+    def test_cannot_repay_booking_not_linked_to_key(self, auth_client):
+        booking = factories.CollectiveBookingFactory()
+        with assert_attribute_does_not_change(booking, "status"):
+            self.repay_booking(auth_client, booking.id, status_code=404)
+
+    def test_can_repay_used_booking(self, auth_client, used_booking):
+        with assert_attribute_value_changes_to(used_booking, "status", models.CollectiveBookingStatus.REIMBURSED):
+            booking_id = used_booking.id
+
+            # 1. get api key
+            # 2. get FF
+            # 3. get collective booking
+            # 4. update booking
+            with assert_num_queries(4):
+                self.repay_booking(auth_client, booking_id, status_code=204)
+
+    def test_cannot_repay_pending_booking(self, auth_client, pending_booking):
+        with assert_attribute_does_not_change(pending_booking, "status"):
+            self.repay_booking(auth_client, pending_booking.id, status_code=403)
```

Comment:
Même remarque qu'[ici](https://github.com/pass-culture/pass-culture-main/pull/14100/files#r1758793507)

```suggestion
            self.repay_booking(auth_client, pending_booking.id, status_code=403)
```